source/material_model/equation_of_state/multicomponent_compressible.cc

```diff
@@ -75,6 +80,9 @@ namespace aspect
       void
       MulticomponentCompressible<dim>::declare_parameters (ParameterHandler &prm)
       {
+        prm.declare_entry ("Use adiabatic pressure for density","false",
```

Comment:
We implemented such a switch for the simple benchmark model we used in the publication about the PDA to illustrate the difference between full and adiabatic pressure, however I do not think this is a good idea for a production model.

Until you can show me that:
1. Using the full pressure for density is stable in a realistic application model. And
2. That you need the full density in order to get an accurate result.

I would oppose even implementing the option for this. This is because using the full pressure simply does not lead to a stable solution.

I am open to be convinced otherwise, but I need to see the data for that. As I discussed with @naliboff this morning, I would actually be surprised if this switch makes a significant difference for your model.